playbooks/roles/install-wetty/tasks/wetty.yml

```diff
@@ -0,0 +1,266 @@
+---
+
+- name: Install required packages
+  yum:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - make
+    - nmap-ncat
+    - npm
+    - gcc-c++
+  become: true
+
+- name: Check if directory exist example
+  stat:
+    path: /etc/letsencrypt/live/{{ openshift_public_hostname }}
+  register: cert_directory
+
+- name: Create directory for certs
+  file: path=/etc/letsencrypt/live/{{ openshift_public_hostname }} state=directory
+  become: true
+  when: not cert_directory.stat.exists
+
+- name: Get SSL cert for wetty
+  command: >-
+    docker run --rm --name certbot
+    -v "/etc/letsencrypt:/etc/letsencrypt:z"
+    -v "/var/lib/letsencrypt:/var/lib/letsencrypt:z"
+    -e AWS_ACCESS_KEY_ID={{ lookup('env', 'AWS_ACCESS_KEY_ID') }}
+    -e AWS_SECRET_ACCESS_KEY={{ lookup('env', 'AWS_SECRET_ACCESS_KEY') }}
+    certbot/dns-route53 certonly
+    --non-interactive
+    --agree-tos
+    --email "{{ cert_email_address }}"
+    --dns-route53
+    --dns-route53-propagation-seconds 30
+    --domain "{{ openshift_public_hostname }}"
+  args:
+    creates: /etc/letsencrypt/live/{{ openshift_public_hostname }}
+  become: true
+  when: letsencrypt_cert_generation and not cert_directory.stat.exists
+
+- name: Get self-signed SSL cert for wetty
+  command: >
+    openssl req -new
+      -x509
+      -nodes
+      -days {{ openssl_days | default(3650) }}
+      -subj "/C={{ openssl_country | default('US') }}/ST={{ openssl_state | default('California') }}/L={{ openssl_ity | default('Los Angeles') }}/O={{ openssl_organization | default('Red Hat, Inc.') }}/OU={{ openssl_unit | default('IT') }}/CN={{ openshift_public_hostname }}/emailAddress={{ cert_email_address }}"
+      -newkey rsa:2048
+      -keyout /etc/letsencrypt/live/{{ openshift_public_hostname }}/privkey.pem
+      -out /etc/letsencrypt/live/{{ openshift_public_hostname }}/cert.pem
+  args:
+    creates: /etc/letsencrypt/live/{{ openshift_public_hostname }}/*.pem
+  become: true
+  when: not letsencrypt_cert_generation and not cert_directory.stat.exists
```

Comment:
When would you wind up needing to create vs. using self-signed, when provisioning this workshop?